### PR TITLE
add support for the updateSecretDesc restful API

### PR DESCRIPTION
**Pull request by Yidingluo1**
this restful API will be exposed to public as:
POST <ehsm_srv_address>/ehsm?Action=UpdateSecretDesc

which is used to update the description of secret.

test: passed the unittest

Signed-off-by: yidingluo1 <luo.yd@neusoft.com>

### CODE REVIEW

File: ehsm_kms_service/secret_manager_apis.js [hunk 1]
```diff
@@ -265,9 +265,69 @@ const createSecret = async (res, appid, payload, DB) => {
...
+//update the description of secret
+const updateSecretDesc = async (res, appid, payload, DB) => {
+    //get and check param in payload
+    let secretName = getParam_String(payload, 'secretName')
+    let description = getParam_String(payload, 'description')
+    if (!checkStringParam(secretName, true, SECRETNAME_LENGTH_MAX)) {
+        res.send(_result(400, 'secretName cannot be empty, must be string and length not more than 64'))
+        return
+    }
+    if (!checkStringParam(description, true, DESCRIPTION_LENGTH_MAX)) {
```
Comment (yang8621) [inline ehsm_kms_service/secret_manager_apis.js:277]: empty description '' still needs support.

Comment (Yidingluo1) [reply ehsm_kms_service/secret_manager_apis.js:277]: ok, it will be add

File: ehsm_kms_service/secret_manager_apis.js [hunk 2]
```diff
@@ -265,9 +265,69 @@ const createSecret = async (res, appid, payload, DB) => {
...
+//update the description of secret
+const updateSecretDesc = async (res, appid, payload, DB) => {
+    //get and check param in payload
+    let secretName = getParam_String(payload, 'secretName')
+    let description = getParam_String(payload, 'description')
+    if (!checkStringParam(secretName, true, SECRETNAME_LENGTH_MAX)) {
+        res.send(_result(400, `secretName cannot be empty, must be string and length not more than ${SECRETNAME_LENGTH_MAX}`))
+        return
+    }
+    if (!checkStringParam(description, false, DESCRIPTION_LENGTH_MAX)) {
```
Comment (yang8621) [inline ehsm_kms_service/secret_manager_apis.js:277]: if the description is **undefined**, what happens?

Comment (Yidingluo1) [reply ehsm_kms_service/secret_manager_apis.js:277]: Same result as ' '

Comment (Yidingluo1) [reply ehsm_kms_service/secret_manager_apis.js:277]: Undefined is equivalent to entering an empty description

File: ehsm_kms_service/secret_manager_apis.js [hunk 3]
```diff
@@ -265,9 +265,69 @@ const createSecret = async (res, appid, payload, DB) => {
...
+        ],
+        limit: 1,
+    }
+    await DB.partitionedFind('secret_metadata', secret_name_query)
+        .then((secret_metadata_res) => {
+            if (secret_metadata_res.docs.length > 0) {
+                secret_metadata_res.docs[0].description = description
+                DB.insert(secret_metadata_res.docs[0])
+                    .then(() => {
+                        res.send(_result(200, 'update secret seccess'))
```
Comment (HawieLong) [inline ehsm_kms_service/secret_manager_apis.js:310]: success

File: test/cli/updateSecretDesc.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import urllib.parse
+
+import _utils_
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--url', type=str, help='the address of the ehsm_kms_server', required=True)
+    parser.add_argument('--secretName', type=str, help='the name of the secret', required=True)
+    parser.add_argument('--description', type=str, help='the description of the secret', required=True)
```
Comment (HawieLong) [inline test/cli/updateSecretDesc.py:19]: if (!checkStringParam(description, false, DESCRIPTION_LENGTH_MAX)) {
description requirement needs changing to false here as well?

Comment (Yidingluo1) [reply test/cli/updateSecretDesc.py:19]: it will be changed

File: ehsm_kms_service/secret_manager_apis.js (hunk 3 above)
Comment (Yidingluo1) [reply ehsm_kms_service/secret_manager_apis.js:310]: it will be changed

File: ehsm_kms_service/secret_manager_apis.js [hunk 5]
```diff
@@ -265,9 +265,69 @@ const createSecret = async (res, appid, payload, DB) => {
...
+            'appid',
+            'secretName',
+            'encryptionKeyId',
+            'description',
+            'createTime',
+            'deleteTime',
+            'plannedDeleteTime',
+            'rotationInterval',
+            'lastRotationDate',
+            'nextRotationDate'
```
Comment (syan10) [inline ehsm_kms_service/secret_manager_apis.js:300]: does it means all the fields will be returned?

Comment (Yidingluo1) [reply ehsm_kms_service/secret_manager_apis.js:300]: yes, need to re insert the fields of the Saved doc

**Review by syan10 — APPROVED** [review]

**Comment by yang8621** [comment]
seems updating the description will not impact versionId, right?   All versions share the same description.

**Comment by Yidingluo1** [comment]
> 

yes

**Comment by Yidingluo1** [comment]
> seems updating the description will not impact versionId, right? All versions share the same description.

yes

**Pull request closed without merging**